services/_internal/maven-EGI/workflowEGI/src/main/java/it/infn/ct/DescribeVM.java

```diff
@@ -0,0 +1,329 @@
+    {
+
+	try {
+		System.out.println();
+
+		if (properties.getProperty("OCCI_RESOURCE_ID").contains("compute")) 
+		{
+			System.out.println("[ VM DESCRIPTION ]");
+				
+			List<Entity> entities = client.describe(URI.create(properties.getProperty("OCCI_RESOURCE_ID")));
+			String[] Attributes = entities.get(0).toText().split(";");
+
+			for (int i=0; i<Attributes.length; i++)
+				{
+					if (Attributes[i].contains("occi.networkinterface.address"))
+					{							
+						publicIP = Attributes[i].replace("occi.networkinterface.address=","");
+					}
+			}			
+
+			for (Entity entity : entities) {
+
+				Map<Attribute, String> map = entity.getAttributes();
+
+				for (Map.Entry<Attribute, String> entry : map.entrySet()) {
+
+					vmState = map.get(new Attribute("occi.compute.state"));}
+
+					if (vmState != null) System.out.println("occi.compute.state = " + vmState);
+
+				}
+		}
+		 
+		
+		if (properties.getProperty("OCCI_RESOURCE_ID").contains("storage")) 
+		{
+			System.out.println("[ STORAGE DESCRIPTION ]");
+			System.out.println("[[ " + properties.getProperty("OCCI_RESOURCE_ID") + " ]]");
+		        List<Entity> entities = client.describe(URI.create(properties.getProperty("OCCI_RESOURCE_ID")));
+
+			String title = "", size = "", state = "", ID = "";
+		        for (Entity entity : entities) {
+				ID = entity.getId();
+				
+				Map<Attribute, String> map = entity.getAttributes();
+				for (Map.Entry<Attribute, String> entry : map.entrySet()) {
+					title = map.get(new Attribute("occi.core.title"));
+					size = map.get(new Attribute("occi.storage.size"));
+					state = map.get(new Attribute("occi.storage.state"));
+					//System.out.println(entry.getKey() + " - " + entry.getValue());
+				}
+
+				System.out.println("\n>> location: " + "/storage/" + ID);
+				System.out.println("occi.core.id = " + ID);
+				System.out.println("occi.core.title = " + title);
+				System.out.println("occi.storage.size = " + size);
+				System.out.println("occi.storage.state = " + state);
+			}
+		}
+
+                if (properties.getProperty("OCCI_RESOURCE_ID").contains("network")) 
+                {
+                        System.out.println("[ NETWORK DESCRIPTION ]");
+                        System.out.println("[[ " + properties.getProperty("OCCI_RESOURCE_ID") + " ]]");
+                        List<Entity> entities = client.describe(URI.create(properties.getProperty("OCCI_RESOURCE_ID")));
+
+                        String title = "", size = "", state = "", ID = "", summary = "", allocation = "";
+			String address = "", vlan = "", netid = "", netvlan = "", phydev = "", bridge = "";
+
+                        for (Entity entity : entities) {
+                                ID = entity.getId();
+
+                                Map<Attribute, String> map = entity.getAttributes();
+                                for (Map.Entry<Attribute, String> entry : map.entrySet()) {
```

Comment:
Code formatting. It is better to make brackets look consistent. Like here you have ")" and "{" on the same line, but most of the time they are on separate lines